webpack/scenes/ContentViews/Create/CreateContentViewForm.js

```diff
@@ -0,0 +1,154 @@
+  const error = useSelector(selectCreateContentViewError);
+
+  useEffect(() => {
+    const { id } = response;
+    if (id && status === STATUS.RESOLVED) {
+      setRedirect(true);
+      setRedirectURL(`/labs/content_views/${id}`);
+    }
+  }, [JSON.stringify(response), status, error]);
+
+  const onSave = () => dispatch(createContentView({
+    name,
+    label,
+    description,
+    composite,
+    dependencies,
+    auto_publish: (autoPublish && composite),
+    import_only: importOnly,
+  }));
+
+  useEffect(
+    () => {
+      setLabel(name.replace(/ /g, '_'));
+    },
+    [name],
+  );
+
+  if (redirect) {
+    return (<Redirect to={redirectUrl} />);
+  }
+
+  return (
+    <Form>
+      <FormGroup label="Name" isRequired fieldId="name">
+        <TextInput
+          isRequired
+          type="text"
+          id="name"
+          name="name"
+          value={name}
+          onChange={value => setName(value)}
+        />
+      </FormGroup>
+      <FormGroup label="Label" isRequired fieldId="label">
+        <TextInput
+          isRequired
+          type="text"
+          id="label"
+          name="label"
+          value={label}
+          onChange={value => setLabel(value)}
+        />
+      </FormGroup>
+      <FormGroup label="Description" fieldId="description">
+        <TextArea
+          isRequired
+          type="text"
+          id="description"
+          name="description"
+          value={description}
+          onChange={value => setDescription(value)}
+        />
+      </FormGroup>
+      <FormGroup isInline fieldId="component">
+        <Radio
+          id="component"
+          name="component"
+          label={labelComponent()}
+          isChecked={component}
+          onChange={(checked) => { setComponent(checked); setComposite(!checked); }}
+          description="Consists of repositories"
+        />
+      </FormGroup>
+      <FormGroup isInline fieldId="composite">
+        <Radio
+          id="composite"
+          name="composite"
+          label={labelComposite()}
+          isChecked={composite}
+          onChange={(checked) => { setComposite(checked); setComponent(!checked); }}
+          description="Consists of more than one component view"
+        />
+      </FormGroup>
+      {!composite &&
+        <FormGroup isInline fieldId="dependencies">
+          <Checkbox
+            id="dependencies"
+            name="dependencies"
+            label={labelDependencies()}
+            isChecked={dependencies}
+            onChange={checked => setDependencies(checked)}
+          />
+        </FormGroup>}
+      {composite &&
+        <FormGroup isInline fieldId="autoPublish">
+          <Checkbox
+            id="autoPublish"
+            name="autoPublish"
+            label={labelAutoPublish()}
+            isChecked={autoPublish}
+            onChange={checked => setAutoPublish(checked)}
+          />
+        </FormGroup>}
+      <FormGroup isInline fieldId="importOnly">
```

Comment:
Would be nice to have a description for this as well

Reply:
Added desc..